execute/tokendata/usdc/http.go

```diff
@@ -0,0 +1,198 @@
+package usdc
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+)
+
+const (
+	apiVersion      = "v1"
+	attestationPath = "attestations"
+
+	// defaultCoolDownDurationSec defines the default time to wait after getting rate limited.
+	// this value is only used if the 429 response does not contain the Retry-After header
+	defaultCoolDownDuration = 5 * time.Minute
+
+	// maxCoolDownDuration defines the maximum duration we can wait till firing the next request
+	maxCoolDownDuration = 10 * time.Minute
+)
+
+type HTTPClient interface {
+	// Get calls the USDC attestation API with the given USDC message hash.
+	// The attestation service rate limit is 10 requests per second. If you exceed 10 requests
+	// per second, the service blocks all API requests for the next 5 minutes and returns an
+	// HTTP 429 response.
+	//
+	// Documentation:
+	//
+	//	https://developers.circle.com/stablecoins/reference/getattestation
+	//	https://developers.circle.com/stablecoins/docs/transfer-usdc-on-testnet-from-ethereum-to-avalanche
+	Get(ctx context.Context, messageHash [32]byte) ([]byte, int, error)
```

Comment:
Rename `Get` -> `GetAttestation`?
Explain what's the `int` we return? or type alias to `type HttpStatus int`